src/actions/courseActions.js

```diff
@@ -0,0 +1,53 @@
+ import * as types from './actionTypes';
+ import CourseApi from '../api/mockCourseApi';
+ import {beginAjaxCall, ajaxCallError} from './ajaxStatusActions';
+
+export const createCourseSuccess=(course)=>{
+    return{type:types.CREATE_COURSE_SUCCESS, course};
+};
+
+export const loadCoursesSuccess = (courses)=> {
+  return {type: types.LOAD_COURSES_SUCCESS, courses};
+};
+
+export const loadCourses = () => {
+    return function (dispatch ) {
+      dispatch(beginAjaxCall());
+      return CourseApi.getAllCourses().then(courses => {
+        dispatch(loadCoursesSuccess(courses));
+      }).catch(error => {
+        throw(error);
+      });
+    };
+  };
+  export const updateCourseSuccess = (course)=> {
+    return {type: types.UPDATE_COURSE_SUCCESS, course};
+  };
+
+    export const saveCourse = (course)=> {
+      return function (dispatch, getState) {
+        dispatch(beginAjaxCall());
+        return CourseApi.saveCourse(course).then(course => {
+          course.id ? dispatch(updateCourseSuccess(course)) : dispatch(createCourseSuccess(course));
+        }).catch(error => {
+
+          throw(error);
+        });
+      };
+  };
+
+  export const  deleteCourse = (course)=> {
+    return function (dispatch, getState) {
+
+      return CourseApi.deleteCourse(course).then(course => {
+        dispatch({
+          type: types.DELETE_COURSE_SUCCESS
+          
+        });
+        dispatch(loadCourses());
```

Comment:
Do you need to call this `loadCourses` since all course is on the state? A typical delete should
1. click on the delete button
2. dispatch an event to delete the course.
3. remove that course from state to using the course id
4 your reducer might handle that like `return courses.filter(course => course.id !== deletedCourceID)`
5. this would update the state removing the deleted course without reloading all courses

Reloading course would be an extra API call and would make more sense if we have a large dataset that is paginated. since we don't  know the nest data, it would make sense to reload the dataset